volatility/framework/automagic/pdbscan.py

```diff
@@ -410,17 +248,19 @@ def check_kernel_offset(self,
         # Scan a few megs of the virtual space at the location to see if they're potential kernels
 
         valid_kernels = {}  # type: ValidKernelsType
```

Comment:
@ikelos the name "valid_kernels" throughout the method_ function names and variable is confusing as each method only will return one kernel and breaks their loops when a valid kernel is found. If we can change the variable names and function names to singular that would make initial reading much easier to grasp.

Reply:
Sure, although I think originally it was supposed to return multiple valid kernels.  It feels like this review was significantly over existing code?  I take it that the changes made by the patch were generally acceptable?

You're right, it's very strange.  I think the original code looked for multiple potential offsets, but given this only allows one per layer name, it doesn't seem all that useful.  I'm wondering whether the methods should return more, perhaps by a generator?  Given the dictionary only ever gets one entry, I've converted it a tuple.  We'd need to rework if if we did ever want the automagic to return multiple options (and we'd have to figure out how to order the methods, whether that be one attempt per method, or all options from each method)...

Reply:
Yes, sorry again about reviewing code not part of the diff, but before today I had never reviewed this file so I needed to read/understand all the functions called to understand what the diff was doing. All the other code in the two pdb* files looks really nice